src/components/buttons/z-button/index.tsx

```diff
@@ -26,42 +32,45 @@ export class ZButton {
   @Prop({ reflect: true }) disabled?: boolean = false;
   /** HTML button type attribute. */
   @Prop() type?: HTMLButtonElement["type"] = ButtonTypeEnum.button;
-  /** Graphical variant: `primary`, `secondary`, `tertiary`, `dark-bg`. Defaults to `primary`. */
-  @Prop({ reflect: true }) variant?: ButtonVariantBean = ButtonVariantEnum.primary;
+  /** Graphical variant: `primary`, `secondary`, `tertiary`. Defaults to `primary`. */
+  @Prop({ reflect: true }) variant?: ButtonVariantBean =
+    ButtonVariantEnum.primary;
   /** `z-icon` name to use (optional). */
   @Prop() icon?: string;
   /** Available sizes: `big`, `small` and `x-small`. Defaults to `big`. */
   @Prop({ reflect: true })
   size?: ButtonSizeEnum = ButtonSizeEnum.big;
-  /** Reduce button size (deprecated).
-   * @deprecated Use `size` prop.
-   */
-  @Prop({ reflect: true }) issmall?: boolean = false;
-  /** Spy to render square button. */
+  /** set label text  */
   @Prop({ reflect: true })
-  square?: boolean = false;
+  text?: string;
 
   render() {
-    this.hostElement.style.pointerEvents = this.disabled ? "none" : "auto";
-
-    return (
-      <slot name="element">
-        <button
+    if (this.href)
```

Comment:
i due tag hanno degli attributi in comune: aria-label, id e class. Si potrebbe evitare di duplicarli creando un metodo che ritorna un oggetto che li contiene e poi richiamarlo destrutturando il valore di ritorno. Una cosa simile viene fatta sullo z-input, metodo `getTextAttributes`